keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,644 @@
+Resource to represent a Bucket in OSP. Buckets are cluster scoped.
+
+```yaml
+Bucket {
+  TypeMeta
+  ObjectMeta
+  
+  Spec BucketSpec {
+    // Provisioner is the name of driver associated with this bucket
+    Provisioner string 
+
+    // DeletionPolicy is used to specify how COSI should handle deletion of this 
+    // bucket. There are 3 possible values:
+    //  - Retain: Indicates that the bucket should not be deleted from the OSP
+    //  - Delete: Indicates that the bucket should be deleted from the OSP
+    //        once all the workloads accessing this bucket are done
+    //  - ForceDelete: Indicates that the bucket and its contents should be 
+    //        deleted from the OSP even if workloads are currently accessing 
+    //        this bucket
+    DeletionPolicy DeletionPolicy 
+    
+    // Name of the BucketClass specified in the BucketRequest
+    // +optional
+    BucketClassName string 
+    
+    // Name of the BucketRequest that resulted in the creation of this Bucket
+    // +optional
+    BucketRequest corev1.ObjectReference
+    
+    // AllowedNamespaces is the list of namespaces from which workloads  
+    // can access this Bucket. If empty, all namespaces can access this access
+    // +optional 
+    AllowedNamespaces []string 
+
+    // Protocol is the data API this bucket is expected to adhere to.
+    // The possible values for protocol are:
+    // -  S3: Indicates Amazon S3 protocol
+    // -  Azure: Indicates Microsoft Azure BlobStore protocol
+    // -  GCS: Indicates Google Cloud Storage protocol
+    Protocol Protocol
+    
+    // Parameters is an opaque map for passing in configuration to a driver
+    // for creating the bucket
+    // +optional
+    Parameters map[string]string 
+  }    
+  
+  Status BucketStatus {
+    // BucketAvailable indicates that the bucket is ready for consumption
+    // by workloads
+    BucketAvailable bool
+    
+    // BucketID is the unique id for the bucket in the OSP
+    BucketID string
+  }
 }
 ```
 
-## ProvisonerCreateBucket
+## BucketRequest
 
-This call is made to create the bucket in the backend. If a bucket that matches both name and parameters already exists, then OK (success) must be returned. If a bucket by same name, but different parameters is provided, then the appropriate error code `ALREADY_EXISTS` must be returned by the provisioner. The call to *ProvisonerCreateBucket* MUST be idempotent.
+A request to create a Bucket in OSP. BucketRequest is a namespaced resource
 
+```yaml
+BucketRequest {
+  TypeMeta
+  ObjectMeta
+  
+  Spec BucketRequestSpec {
+    // Protocol is the data API this bucket is expected to adhere to.
+    // The possible values for protocol are:
+    // -  S3: Indicates Amazon S3 protocol
+    // -  Azure: Indicates Microsoft Azure BlobStore protocol
+    // -  GCS: Indicates Google Cloud Storage protocol
+    Protocol Protocol
+    
+    // Name of the BucketClass 
+    // +optional    
+    BucketClassName string
+  }
```

Comment:
What is the reason for not having a bucket name prefix field, like in the earlier KEP? A prefix, which could be in the BC instead of the BR, allows a bucket name pattern to be established, which could be useful to the backend object store admin.

Reply:
The BR does not reference a Bucket so how can COSI get the name of the B from a BAR?
Note: the BAR also does not reference a B.

Reply:
> The BR does not reference a Bucket so how does COSI get the name of the B from a BAR?
Note: the BAR also does not reference a B.

Will fix both

Reply:
Is there no use for a BR.spec.BucketPrefix field, like in earlier drafts?